src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -178,10 +178,16 @@ internal HttpClient GetHttpClient()
                         switch (_authenticationScheme)
                         {
                             case AuthenticationSchemes.Basic:
-                                if (credentials.UserName.UserName != string.Empty)
+                                if (credentials.UserName.UserName == null)
                                 {
-                                    creds = new NetworkCredential(credentials.UserName.UserName, credentials.UserName.Password);
+                                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("userName");
                                 }
+                                if (credentials.UserName.UserName == string.Empty)
```

Comment:
On desktop,
1) Empty string, It will throw ArgumentException.
2) String containing only whitespace. It's treated as non empty string and will pass validation.

As for all other case statements, I am not clear about the usage scenarios and we do not have test cases failing because of them. I would suggest we solve them separately.